src/core/tests/type_prop/unique.cpp

```diff
@@ -0,0 +1,163 @@
+    ASSERT_OUTPUT_SHAPES(unique, {{PartialShape{1}, PartialShape{1}, PartialShape{1}, PartialShape{1}}});
+}
+
+TEST(type_prop, unique_3d_scalar_axis) {
+    const auto data = make_shared<opset10::Parameter>(element::f32, PartialShape{2, 4, 2});
+    const auto axis = make_shared<opset10::Constant>(element::i32, Shape{}, 1);
+    const auto unique = make_shared<opset10::Unique>(data, axis);
+
+    ASSERT_ELEMENT_TYPES(unique, {{element::f32, element::i64, element::i64, element::i64}});
+    ASSERT_OUTPUT_SHAPES(
+        unique,
+        {{PartialShape{{2}, {1, 4}, {2}}, PartialShape{{1, 16}}, PartialShape{{1, 16}}, PartialShape{{1, 16}}}});
+}
+
+TEST(type_prop, unique_3d_axis_1d) {
+    const auto data = make_shared<opset10::Parameter>(element::f32, PartialShape{2, 4, 2});
+    const auto axis = make_shared<opset10::Constant>(element::i32, Shape{1}, 2);
+    const auto unique = make_shared<opset10::Unique>(data, axis);
+
+    ASSERT_ELEMENT_TYPES(unique, {{element::f32, element::i64, element::i64, element::i64}});
+    ASSERT_OUTPUT_SHAPES(
+        unique,
+        {{PartialShape{{2}, {4}, {1, 2}}, PartialShape{{1, 16}}, PartialShape{{1, 16}}, PartialShape{{1, 16}}}});
+}
+
+TEST(type_prop, unique_3d_negative_axis) {
+    const auto data = make_shared<opset10::Parameter>(element::f32, PartialShape{2, 4, 2});
+    const auto axis = make_shared<opset10::Constant>(element::i64, Shape{}, -3);
+    const auto unique = make_shared<opset10::Unique>(data, axis);
+
+    ASSERT_ELEMENT_TYPES(unique, {{element::f32, element::i64, element::i64, element::i64}});
+    ASSERT_OUTPUT_SHAPES(
+        unique,
+        {{PartialShape{{1, 2}, {4}, {2}}, PartialShape{{1, 16}}, PartialShape{{1, 16}}, PartialShape{{1, 16}}}});
+}
+
+TEST(type_prop, unique_dynamic_dim_at_axis) {
+    const auto data = make_shared<opset10::Parameter>(element::f32, PartialShape{2, -1, 2});
+    const auto axis = make_shared<opset10::Constant>(element::i64, Shape{}, 1);
+    const auto unique = make_shared<opset10::Unique>(data, axis);
+
+    ASSERT_ELEMENT_TYPES(unique, {{element::f32, element::i64, element::i64, element::i64}});
+    ASSERT_OUTPUT_SHAPES(unique,
+                         {{PartialShape{{2}, {-1}, {2}}, PartialShape{{-1}}, PartialShape{{-1}}, PartialShape{{-1}}}});
+}
+
+TEST(type_prop, unique_dynamic_rank) {
+    const auto data = make_shared<opset10::Parameter>(element::f32, PartialShape::dynamic());
+    const auto axis = make_shared<opset10::Constant>(element::i32, Shape{}, 1);
+    const auto unique = make_shared<opset10::Unique>(data, axis);
+
+    ASSERT_ELEMENT_TYPES(unique, {{element::f32, element::i64, element::i64, element::i64}});
+    ASSERT_OUTPUT_SHAPES(unique,
+                         {{PartialShape::dynamic(), PartialShape{{-1}}, PartialShape{{-1}}, PartialShape{{-1}}}});
+}
+
+TEST(type_prop, unique_dynamic_rank_negative_axis) {
+    const auto data = make_shared<opset10::Parameter>(element::f32, PartialShape::dynamic());
```

Comment:
Add test when PartialShape::dynamic(X) and when Partial shape is {Dimension(2,3), y, z} Add different combination of interval dimension.